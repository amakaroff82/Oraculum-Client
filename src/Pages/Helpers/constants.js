import { MC_BASE_URL } from '../../Shared/constants';

