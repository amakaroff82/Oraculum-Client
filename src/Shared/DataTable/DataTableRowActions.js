import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';

import DataTableRowAction from './DataTableRowAction';

const styles = {
  root: {
    display: 'block',
  },
};

export const DataTableRowActions = ({ actions, classes }) => {
  return (
    <div className={classes.root}>
      {actions &&
        actions.map((action, index) => {
          return <DataTableRowAction key={index} {...action} />;
        })}
    </div>
  );
};

DataTableRowActions.propTypes = {
  actions: PropTypes.array.isRequired,
};

export default withStyles(styles, { name: 'DataTableRowActions' })(
  DataTableRowActions
);
