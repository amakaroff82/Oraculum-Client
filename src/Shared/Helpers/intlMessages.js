export const intlMessages = {
  orchestration: { id: 'Orchestration', defaultMessage: 'Orchestration' },
  products: { id: 'Products', defaultMessage: 'Products' },
  status: { id: 'Status', defaultMessage: 'Status' },
  dateLastModified: {
    id: 'DateLastModified',
    defaultMessage: 'Date last modified',
  },
  title: { id: 'Title', defaultMessage: 'Titre' },
  topic: { id: 'Topic', defaultMessage: 'Topic' },
  createdBy: { id: 'CreatedBy', defaultMessage: 'Created by' },
  version: { id: 'Version', defaultMessage: 'Version' },
};

export default intlMessages;
