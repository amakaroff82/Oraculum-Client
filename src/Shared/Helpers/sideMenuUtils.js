import React from 'react';
import { ListItem, ListItemText } from 'material-ui/List';

export const renderListItems = (listItemData, listAction, identifierKey) => {
  if (listItemData && listAction && identifierKey) {
    return listItemData.map((element, index) => {
      return (
        <ListItem
          key={element[identifierKey]}
          button
          onClick={() => listAction(element)}
        >
          <ListItemText color="primary" primary={element[identifierKey]} />
        </ListItem>
      );
    });
  }
};
