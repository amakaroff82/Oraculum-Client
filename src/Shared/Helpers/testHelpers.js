// Create mock response for testing
export const makeMockResponse = (
  response,
  status = 200,
  statusText = 'OK',
  headers = { 'Content-Type': 'application/json' }
) => {
  const init = {
    status,
    statusText,
    headers,
  };
  return new window.Response(JSON.stringify(response), init);
};

export const jsonParseError = () => {
  try {
    JSON.parse('+data+');
  } catch (err) {
    return err;
  }
};

export const makeFakeError = (
  response = null,
  status = 400,
  statusText = 'Bad Request'
) => {
  const error = new Error('Error');
  error.response = makeMockResponse(response, status, statusText);
  return error;
};
