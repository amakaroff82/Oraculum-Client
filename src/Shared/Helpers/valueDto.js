/// Create items for binding Std ValueDTOs from the web service
export const getDropDownValues = dtoArray => {
  if (!dtoArray) {
    return [];
  }
  return dtoArray.map(element => ({
    value: element.code,
    label: element.description || element.code,
  }));
};

export const getAutocompleteValues = dtoArray => {
  if (!dtoArray) {
    return [];
  }
  var result = {};
  for (var value of dtoArray) {
    result[value['code']] = value['description'] || value['code'];
  }
  return result;
};
