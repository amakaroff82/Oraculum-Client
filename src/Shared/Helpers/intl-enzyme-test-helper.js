/**
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { intlMessages } from './intlMessages';

