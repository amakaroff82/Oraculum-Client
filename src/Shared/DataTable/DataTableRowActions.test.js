import * as React from 'react';
import { shallowWithIntl } from '../../Shared/Helpers/intl-enzyme-test-helper';
import toJson from 'enzyme-to-json';
import { DataTableRowActions } from './DataTableRowActions';

const defaultProps = {
  classes: {},
  actions: [],
};

const setup = setupProps => {
  const props = { ...defaultProps, ...setupProps };
  return shallowWithIntl(<DataTableRowActions {...props} />);
};

describe('<DataTableRowActions />', () => {
  it('renders snapshot correctly with required props', () => {
    const tree = setup();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders snapshot correctly with a child action', () => {
    const altProps = {
      actions: [
        {
          childComponent: <div />,
        },
      ],
    };
    const tree = setup(altProps);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
