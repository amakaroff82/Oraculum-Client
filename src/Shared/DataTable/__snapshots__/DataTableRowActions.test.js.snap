// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTableRowActions /> renders snapshot correctly with a child action 1`] = `
<div>
  <withStyles(Component)
    childComponent={<div />}
  />
</div>
`;

exports[`<DataTableRowActions /> renders snapshot correctly with required props 1`] = `<div />`;
